src/cryptography/nostr_bech32.cpp

```diff
@@ -0,0 +1,743 @@
+
+    int ret;
+    ret = put_byte(&cur, &type);
+    if (!ret) {
+        fprintf(stderr, "Error executing 'put_byte'\n");
+        return ret;
+    }
+
+    ret = put_byte(&cur, &len);
+    if (!ret) {
+        fprintf(stderr, "Error executing 'put_byte'\n");
+        return ret;
+    }
+
+    ret = put_bytes(&cur, strlen(tag), (uint8_t *)tag);
+    if (!ret) {
+        fprintf(stderr, "Error executing 'put_bytes'\n");
+        return ret;
+    }
+
+    if (nb_relays > 0) {
+        for (int i = 0; i < nb_relays;i++) {
+            uint8_t type = TLV_RELAY;
+            uint8_t len = strlen(relays[i]);
+
+            ret = put_byte(&cur, &type);
+            if (!ret) {
+                fprintf(stderr, "Error executing 'put_byte'\n");
+                return ret;
+            }
+
+            ret = put_byte(&cur, &len);
+            if (!ret) {
+                fprintf(stderr, "Error executing 'put_byte'\n");
+                return ret;
+            }
+
+            ret = put_bytes(&cur, len, (uint8_t *)relays[i]);
+            if (!ret) {
+                fprintf(stderr, "Error executing 'put_bytes'\n");
+                return ret;
+            }
+        }
+    }
+
+    type = TLV_AUTHOR;
+    len = KEY_LENGTH;
+    ret = put_byte(&cur, &type);
+    if (!ret) {
+        fprintf(stderr, "Error executing 'put_byte'\n");
+        return ret;
+    }
+
+    ret = put_byte(&cur, &len);
+    if (!ret) {
+        fprintf(stderr, "Error executing 'put_byte'\n");
+        return ret;
+    }
+    for(int i=0;i<KEY_LENGTH;i++) {
+        strncpy(current_byte, pubkey + 2*i, 2);
+        placeholder[i] = (uint8_t)strtol(current_byte, NULL, 16);
+    }
+    ret = put_bytes(&cur, KEY_LENGTH, placeholder);
+    if (!ret) {
+        fprintf(stderr, "Error executing 'put_bytes'\n");
+        return ret;
+    }
+
+    type = TLV_KIND;
+    len = 4;
+    ret = put_byte(&cur, &type);
+    if (!ret) {
+        fprintf(stderr, "Error executing 'put_byte'\n");
+        return ret;
+    }
+
+    ret = put_byte(&cur, &len);
+    if (!ret) {
+        fprintf(stderr, "Error executing 'put_byte'\n");
+        return ret;
+    }
+
+    uint8_t *kind_bytes = (uint8_t*)malloc(len);
+    encode_u32_tlv(kind, kind_bytes);
+
+    ret = put_bytes(&cur, len, kind_bytes);
+
+    if (!ret) {
+        fprintf(stderr, "Error executing 'put_bytes'\n");
+        return ret;
+    }
+    free(kind_bytes);
+
+    uint8_t data[ouput_length];
+    size_t datalen = 0;
+    ret = bech32_convert_bits(data, &datalen, TO_BITS, input_hex, input_len, FROM_BITS, 1);
+    if (!ret) {
+        fprintf(stderr, "Error executing 'bech32_convert_bits'\n");
+        return ret;
+    }
+
+    ret = bech32_encode(naddr, "naddr", data, datalen, MAX_ENCODING_LENGTH, BECH32_ENCODING_BECH32);
+    if (!ret) {
+        fprintf(stderr, "Error executing 'bech32_encode'\n");
+        return ret;
+    }
+
+    return 1;
+
+}
+
+
+int encode_nostr_bech32_nrelay(struct cursor *cur, struct bech32_nrelay *obj) {
+
+}
```

Comment:
This should either be defined, or `nrelay`-related definitions should be removed entirely.